alvr/xtask/src/ci.rs

```diff
@@ -44,21 +44,20 @@ pub fn clippy_ci() {
     let stream = Deserializer::from_slice(&out.stdout).into_iter::<Value>();
 
     // https://doc.rust-lang.org/cargo/reference/external-tools.html#json-messages
-    for message in stream.filter_map(|msg| {
-        let msg = msg.unwrap();
+    let diagnostic_messages = stream
+        .filter_map(|msg| {
+            let msg = msg.unwrap();
 
-        if msg.get("reason")? == "compiler-message" {
-            msg.get("message").map(|x| x.to_owned())
-        } else {
-            None
-        }
-    }) {
-        let msg: CompilerMessage = json::from_value(message).unwrap();
-
-        if msg.message_type != "diagnostic" {
-            continue;
-        }
+            if msg.get("reason")? == "compiler-message" {
+                let msg: CompilerMessage = json::from_value(msg.get("message")?.clone()).ok()?;
+                (msg.message_type != "diagnostic").then_some(msg)
```

Comment:
this seems exactly inverted? should match if it **is** a diagnostic

Reply:
right...